### ISSUE TITLE
Bug: Adding import to smart contract causes deployment to fail

### ISSUE THREAD

**Issue by rpc5255** [iotaledger/wasp-legacy#3177]
**Describe the bug**
I am trying to deploy a smart contract that is written in Go on a private chain. It passes the testing library and I'm able to build it using ```schema -go -build```. When I try to deploy the smart contract, I receive the following error: 
```Error: cannot instantiate: unknown import: `gojs::runtime.ticks` has not been defined```
My smart contract imports the ```encoding/json``` package. When I remove this import from the smart contract, I am able to deploy it to the chain.
 
**To Reproduce**
We can only investigate bug reports that can easily be reproduced and all information needed to reproduce this issue has been supplied with this issue. Please provide the steps to reproduce this issue in the easiest way possible:

1. Go to existing smart contract ```contracts/wasm/inccounter/go/inccounterimpl/funcs.go``` 
2.  Add ```_encoding/json``` to the imports ( ```_``` is to bypass any building errors from an unused import)
3. Build the smart contract using ```schema -go -build```
4. Test the smart contract by going into ```contracts/wasm/inccounter/test``` and running ```go test```. It should pass.
6. Deploy the smart contract on a private chain using ```./wasp-cli chain deploy-contract wasmtime inccounter "inccounter SC" ./contracts/wasm/inccounter/go/pkg/inccounter_go.wasm```
7.  ```Error: cannot instantiate: unknown import: `gojs::runtime.ticks` has not been defined``` should appear

**Expected behavior**
I expected the smart contract to be deployed since it passed the tests. Am I missing something regarding the imports?

**Screenshots**

**Network and versioning**
 - Type of L1 network: Private Tangle
 - Type of Wasp chain: Private Chain
 - Interaction method: Wasp-cli
 - Interaction software: wasp-cli
 - Version of Wasp: v1.0.1-rc.15

**Additional context**

**Comment by rpc5255** [comment]
Response from @BugFreeSoftware in Discord: 

You need to realize that a SC cannot use every library out there. SCs have no way to interact with the outside world other than what is provided by the VM. That means no file access, for example. No timing functions. No multithreading. No pure randomizers.  A SC needs to be 100% deterministic, and it cannot be that with things that may cuase different results between different nodes. They all need to come to the same outcome. That is why a SC function call needs to be provided with all data it needs to safely process. 
The Go-to-Wasm compiler will try to call some functionality that could be available in a browser Wasm VM as external dynamically linked entry points. But our ISC Wasm VM does not provide most of those calls. That means that upon deployment the external functions the Wasm code tries to link to are simply not available and the deployment will fail in that case. 

**TLDR: Don't try to import outside packages into the smart contract.** 

I was using `encoding/json` to convert my internal array struct that was generated by schema into a `String` so that `wasp-cli decode` could be used. `wasp-cli decode` cannot be used in this case because no interface has been defined between `schema` and `wasp-cli`. If you want to serialize the result from a `view`, use the `WasmClient` library to build your own client that can call the SC functions.  See `packages/wasmvm/wasclient/go/test` for an example.
